docker/batch-single/scripts/main.py

```diff
@@ -0,0 +1,734 @@
+
+            #publish message 
+            response = sns.publish(
+                    TopicArn=self.aws_sns_topic,
+                    Message=message
+                )
+        except ClientError as e:
+            logging.error(e)
+
+
+    def _publish_failure_message(self, message):
+        """Helper function that will publish failure message to SNS if an unrecoverable error
+        occurs before processing begins.
+        param: str message: The failure message to publish
+        """
+        heading = "The job {0} for the task {1} submission {2} failed against {3} validation the with" \
+            " the following error: {4}".format(self.job_id, self.s3_submission_task, self.s3_submission_archive, 
+                self.s3_submission_validation_descr, message)
+        self._publish_sns_message(heading)
+
+
+    def _create_sqs_queue(self):
+        """Creates SQS FIFO queue with specified name.
+        :return: The SQS queue url
+        :rtype: str
+        :raises ClientError: SQS client exception  
+        :raises Exception: SQS queue was unable to be created
+        """
+        sqs_client = self.session.client('sqs')
+        queue_name = self.job_id + '.fifo'
+
+        try:
+            logging.info("Creating sqs queue %s", queue_name)
+            queue = sqs_client.create_queue(
+                QueueName = queue_name,
+                Attributes =  {
+                    'MaximumMessageSize': '262144',
+                    'VisibilityTimeout': '3600',
+                    'MessageRetentionPeriod':'1209600',
+                    'FifoQueue': 'true',
+                    'ContentBasedDeduplication': 'true'
+                }   
+            )
+
+            if not queue['QueueUrl']:
+                raise ValueError("Unable to create SQS queue with given name {0}, QueueUrl is empty".format(queue_name))
+
+            return queue['QueueUrl']
+
+        except ClientError as e:
+            err = "Unable to create SQS queue with given name due to boto exception: {0}".format(e)
+            self._publish_failure_message(err)
+            logging.error(err)
+            raise
+        except ValueErrors as e:
+            self._publish_failure_message(e)
+            logging.error(e)
+            raise
+
+
+    def _enqueue_files(self, queue_url, objects):
+        """Uploads all turtle (ttl) files in source directory to the provided S3 bucket,
+        adds each S3 object path as a message on SQS and creates / updates the source log on S3. 
+        The source log is a serialized list of each S3 object path that will
+        be used for processing validation. After all messages have processed and added to 
+        the queue, a final source file will be uploaded with a suffix of '.queued' and the old
+        source file will be removed from S3.
+        :param str bucket_direcotry: The bucket and prefix directory that contains the unprocessed 
```

Comment:
parameter bucket_directory doesn't exist.  Should be replaced with objects and updated to refer to a list of buckets.

Reply:
Updated documentation.